client/service/src/builder.rs

```diff
@@ -529,8 +531,19 @@ where
 		)
 	};
 
-	let rpc = start_rpc_servers(&config, gen_rpc_module, rpc_id_provider)?;
-	let rpc_handlers = RpcHandlers(Arc::new(gen_rpc_module(sc_rpc::DenyUnsafe::No)?.into()));
+	let (rpc, socket_addr) = start_rpc_servers(&config, gen_rpc_module, rpc_id_provider)?;
+
+	//Convert SocketAddr to transport-agnostic MultiAddr
+	let mut multi_addr = Multiaddr::empty().with(Protocol::Tcp(socket_addr.port()));
+	let multi_addr = match socket_addr.ip() {
+		IpAddr::V4(addr) => multi_addr.with(Protocol::Ip4(addr)),
+		IpAddr::V6(addr) => multi_addr.with(Protocol::Ip6(addr)),
+	};
```

Comment:
```suggestion
	let multi_addr = Multiaddr::from(socket_addr);
```

should be identical according to `impl From<IpAddr> for Multiaddr {` in crate `multiaddr`

Reply:
That is a lot less code 🙂